### ISSUE TITLE
10-bit i2c addressing is missing from pico-sdk, but does in fact work.

### ISSUE THREAD

**Issue by zbaltzer228** [raspberrypi/pico-sdk#720]
Over at my [repos readme](https://github.com/zbaltzer228/pico-sdk-i2c-10bit-explained/blob/develop/README.md) I explore and explain how 10-bit i2c addressing is fully supported on the RP2040 but the RP2040 datasheet suggests otherwise in the case of slave mode operation. I have another [forked repo](https://github.com/zbaltzer228/pico-sdk) with just the updates required to update the SDK to fully support 10-bit i2c addressing.

Basically I left all existing functions alone (obviously), and added about 13 new functions with identical names to the existing i2c functions but with a _10bit qualifier on the end. 

But I should ask before doing so, why was 10-bit i2c addressing left out of the SDK ? Was it simply because it's not used much?

If you all decide it's something worth adding to the SDK, I'll open a PR against this issue.
